### PR TITLE
QT refactor

### LINKED ISSUE #44: Refactor question tagging
Question tagging needs to be refactored to use a new collection name. Logicly it should remain mostly unchanged.

### PR DESCRIPTION
**Pull request by BenSegal855**
Closes #44. The `question` and `tagquestion` commands now take args in a much friendlier way utilizing course info from the context of the message. Also, collection names were moved to the config because I had to change one and it was a pain so I thought I would make it easier to change if necessary in the future.

### CODE REVIEW

**Review by renross — APPROVED** [review]

File: src/commands/configuration/addcourse.ts [hunk 1]
```diff
@@ -1,9 +1,9 @@
 import { Message, OverwriteResolvable, Guild, TextChannel } from 'discord.js';
 import { Course } from '@lib/types/Course';
 import { adminPerms } from '@lib/permissions';
-import { GUILDS, ROLES } from '@root/config';
+import { DB, GUILDS, ROLES } from '@root/config';
 
-export const description = 'Creates a corses category and adds all nessary channels/roles.';
+export const description = 'Creates a corses category and adds all necessary channels/roles.';
```
Comment (joshualyon7) [inline src/commands/configuration/addcourse.ts:6]:
```
Courses*
```

File: src/commands/partial visibality question/anonymous.ts [hunk 2]
```diff
@@ -61,8 +61,8 @@ export async function argParser(msg: Message, input: string): Promise<[Course, s
 	} else {
 		const inputtedCourse = courses.find(c => c.name === input.split(' ')[0]);
 		if (!inputtedCourse) {
-			throw `I wasn't able to determine your course baised off of your enrollment or your input. Please specify the corse at the begining of your question.
-Avaliable corses: \`${courses.map(c => c.name).join('`, `')}\``;
+			throw `I wasn't able to determine your course biased off of your enrollment or your input. Please specify the corse at the beginning of your question.
```
Comment (joshualyon7) [inline src/commands/partial visibality question/anonymous.ts:64]: `based*` `course*`

File: src/commands/partial visibality question/anonymous.ts [hunk 3]
```diff
@@ -61,8 +61,8 @@ export async function argParser(msg: Message, input: string): Promise<[Course, s
...
+Available corses: \`${courses.map(c => c.name).join('`, `')}\``;
```
Comment (joshualyon7) [inline src/commands/partial visibality question/anonymous.ts:65]: `courses*`

File: src/commands/partial visibality question/private.ts [hunk 4]
```diff
@@ -53,8 +53,8 @@ export async function argParser(msg: Message, input: string): Promise<[Course, s
 	} else {
 		const inputtedCourse = courses.find(c => c.name === input.split(' ')[0]);
 		if (!inputtedCourse) {
-			throw `I wasn't able to determine your course baised off of your enrollment or your input. Please specify the corse at the begining of your question.
-Avaliable corses: \`${courses.map(c => c.name).join('`, `')}\``;
+			throw `I wasn't able to determine your course biased off of your enrollment or your input. Please specify the corse at the beginning of your question.
```
Comment (joshualyon7) [inline src/commands/partial visibality question/private.ts:56]: `based*` `course*` (same string as before)

File: src/commands/question tagging/question.ts [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+		course = courses.find(c => c.name === user.courses[0]);
+		if (input.startsWith(course.name)) {
+			assignment = input.slice(course.name.length).trim();
+		} else {
+			assignment = input;
+		}
+	} else {
+		const inputtedCourse = courses.find(c => c.name === input.split(' ')[0]);
+		if (!inputtedCourse) {
+			throw 'I wasn\'t able to determine your course biased off of your enrollment or your input. Please specify the corse at the beginning of your question.' +
```
Comment (joshualyon7) [inline src/commands/question tagging/question.ts:70]: `course*`

**Review by joshualyon7 — CHANGES_REQUESTED** [review]
Spelling stuff

**Review by joshualyon7 — APPROVED** [review]

**Pull request merged**